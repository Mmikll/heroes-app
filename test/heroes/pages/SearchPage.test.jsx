const { render, screen, fireEvent } = require("@testing-library/react")
const { MemoryRouter, useNavigate } = require("react-router-dom")
const { SearchPage } = require("../../../src/heroes/pages/SearchPage")

const mockUseNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate : () => mockUseNavigate
}
))

    beforeEach(() => jest.clearAllMocks())

describe('<SearchPage/> test', () => { 
    
    test('should show default', () => { 

        const {container } = render(

            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot()
    })

    test('should show batman and query with input', () => { 

        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        )

        const input = screen.getByRole( 'textbox' )
        expect(input.value).toBe('batman')

        const img = screen.getByRole('img')

        expect(img.src).toContain('/assets/heroes/dc-batman.jpg')

        const div = screen.getByLabelText('noFound')

        screen.debug()
        expect(div.style.display).toBe('none')
    })


    test('should display error div if hero was no found', () => { 

        render(
            <MemoryRouter initialEntries={['/search?q=batm123']}>
                <SearchPage/>
            </MemoryRouter>
        )

        const div = screen.getByLabelText('noFound')

        expect(div.style.display).toBe('')
    })

    test('should call navigate to a new page', () => { 

        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>
        )

        const imput = screen.getByRole('textbox')

        fireEvent.change(imput, {target: {name : 'searchText', value: 'superman' }} )

        const form = screen.getByLabelText('form')
        
        fireEvent.submit(form)

        expect( mockUseNavigate).toHaveBeenCalled()

    })

})