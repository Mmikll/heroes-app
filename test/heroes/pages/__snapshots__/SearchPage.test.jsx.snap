// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage/> test should show default 1`] = `
<div>
  <h1>
    Find Your Hero
  </h1>
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Spider-Man"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2 fa fa-search"
        />
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Start Searching
      </div>
      <div
        aria-label="noFound"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        <b />
         was not found
      </div>
    </div>
  </div>
</div>
`;
